agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -257,6 +257,13 @@ private static void logConfigurationWarnings(Configuration config) {
       configurationLogger.warn(
           "\"clientsecret\" typed of AAD authentication has been deprecated since 3.5.0 GA. Please use \"user-assigned identity\" or \"system-assigned identity\" instead.");
     }
+    if (config.sampling.percentage != null && config.sampling.requestsPerSecond != null) {
+      configurationLogger.warn(
+          "Sampling \"requestsPerSecond\" and \"percentage\" should not be used at the same time."
+              + " Please remove one of them.");
+      config.sampling.percentage = null; // requestsPerSecond takes priority
+    }
```

Comment:
this was missing from 3.4.0+

don't want to add this validation to `Configuration`, since we can only throw exception there, and someone could have misconfigured in the meantime

Reply:
not sure if that is a valid customer scenario though.

Reply:
it's not, but don't want to fail the agent startup suddenly with new release if they had misconfigured it (but it was working) previously